File: ignisign-android/ignisign/src/main/java/com/ignisign/ignisign/public/DocumentEntities.kt
package com.ignisign.ignisign.public

import com.ignisign.ignisign.IgnisignApplicationEnv
import java.util.*

enum class IgnisignDocumentType {
    PDF,
    FILE,
    DATA_JSON,
    PRIVATE_FILE
}

enum class IgnisignDocumentStatus {
    CREATED,
    DOCUMENT_REQUEST,
    PROVIDED,
    ARCHIVED
}

enum class GetPrivateFileErrors {
    DOCUMENT_HASH_DOES_NOT_MATCH_PROVIDED_ONE,
    NOT_AUTHORIZED_TO_GET,
    CANNOT_GET_FILE
}

data class IgnisignDocument(
    val _id: String? = null,
    val appId: String,
    val appEnv: IgnisignApplicationEnv,
    val documentNature: IgnisignDocumentType,
    val status: IgnisignDocumentStatus,
    val documentHash: String,
    val signatureRequestId: String,
    val documentRequestId: String? = null,
    val externalId: String? = null,
    val label: String? = null,
    val description: String? = null,
    val fileName: String? = null,
    val fileSize: Number? = null,
    val mimeType: String? = null,
    val dataJsonContent: String? = null,
    val relatedDocumentId:  String? = null,
    val relatedDocumentType: IgnisignDocumentType? = null,
    val _createdAt: Date? = null
)

data class IgnisignDocumentContainer(
    val document: IgnisignDocument
)

data class IgnisignDocumentContext(
    val ignisignDocument: IgnisignDocument,
    val statements: List<IgnisignSignatureRequestStatement>? = null,
    val documentRequest: IgnisignDocumentRequest? = null,
    val signatureSummaries: List<SignatureSummary>
) {
    data class SignatureSummary(
        val signatureId: String,
        val signerId: String,
        val date: Date
    )
}

data class IgnisignDocumentInitializationDto(
    val signatureRequestId: String,
    val externalId: String? = null,
    val label: String? = null,
    val description: String? = null
)

data class IgnisignDocumentUpdateDto(
    val externalId: String? = null,
    val label: String? = null,
    val description: String? = null
)

File: ignisign-ios/Public/DocumentEntities.swift
//
//  DocumentEntities.swift
//  ignisign-ios
//
//  Created by Marc Nigdélian on 19/12/2023.
//

import Foundation

enum IgnisignDocumentType {
    case PDF
    case FILE
    case DATA_JSON
    case PRIVATE_FILE
}

enum IgnisignDocumentStatus {
    case CREATED
    case DOCUMENT_REQUEST
    case PROVIDED
    case ARCHIVED
}

enum GetPrivateFileErrors {
    case DOCUMENT_HASH_DOES_NOT_MATCH_PROVIDED_ONE
    case NOT_AUTHORIZED_TO_GET
    case CANNOT_GET_FILE
}

struct IgnisignDocument {
    var _id: String
    var appId: String
    var appEnv: IgnisignApplicationEnv
    var documentNature: IgnisignDocumentType
    var status: IgnisignDocumentStatus
    var documentHash: String
    var signatureRequestId: String
    var documentRequestId: String
    var externalId: String
    var label: String
    var description: String
    var fileName: String
    var fileSize: String
    var mimeType: String
    var dataJsonContent: String
    var relatedDocumentId: String
    var relatedDocumentType: String
    var createdAt: Date
}

struct IgnisignDocumentContainer {
    var document: IgnisignDocument
}

struct IgnisignDocumentContext {
    var ignisignDocument: IgnisignDocument
    //var statements: [
}

/*
 data class IgnisignDocumentContext(
     val ignisignDocument: IgnisignDocument,
     val statements: List<IgnisignSignatureRequestStatement>? = null, // Type à définir
     val documentRequest: IgnisignDocumentRequest? = null, // Type à définir
     val signatureSummaries: List<SignatureSummary>
 ) {
     data class SignatureSummary(
         val signatureId: String,
         val signerId: String,
         val date: Date
     )
 }

 data class IgnisignDocumentInitializationDto(
     val signatureRequestId: String,
     val externalId: String? = null,
     val label: String? = null,
     val description: String? = null
 )

 data class IgnisignDocumentUpdateDto(
     val externalId: String? = null,
     val label: String? = null,
     val description: String? = null
 )


 */
